samples/Azure.ResourceManager.Sample/Generated/VirtualMachineScaleSetVirtualMachineExtensionCollection.cs

```diff
@@ -293,20 +294,25 @@ public async virtual Task<Response<bool>> CheckIfExistsAsync(string vmExtensionN
         /// <summary> The operation to get all extensions of an instance in Virtual Machine Scaleset. </summary>
         /// <param name="expand"> The expand expression to apply on the operation. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual Response<IReadOnlyList<VirtualMachineScaleSetVirtualMachineExtension>> GetAll(string expand = null, CancellationToken cancellationToken = default)
+        /// <returns> A collection of <see cref="VirtualMachineScaleSetVirtualMachineExtension" /> that may take multiple service requests to iterate over. </returns>
+        public virtual Pageable<VirtualMachineScaleSetVirtualMachineExtension> GetAll(string expand = null, CancellationToken cancellationToken = default)
         {
-            using var scope = _clientDiagnostics.CreateScope("VirtualMachineScaleSetVirtualMachineExtensionCollection.GetAll");
-            scope.Start();
-            try
+            Page<VirtualMachineScaleSetVirtualMachineExtension> FirstPageFunc(int? pageSizeHint)
             {
-                var response = _virtualMachineScaleSetVMExtensionsRestClient.List(Id.SubscriptionId, Id.ResourceGroupName, Id.Parent.Name, Id.Name, expand, cancellationToken);
-                return Response.FromValue(response.Value.Value.Select(value => new VirtualMachineScaleSetVirtualMachineExtension(Parent, value)).ToArray() as IReadOnlyList<VirtualMachineScaleSetVirtualMachineExtension>, response.GetRawResponse());
-            }
-            catch (Exception e)
-            {
-                scope.Failed(e);
-                throw;
+                using var scope = _clientDiagnostics.CreateScope("VirtualMachineScaleSetVirtualMachineExtensionCollection.GetAll");
```

Comment:
This code is different from the previous one semantically. Previously, user would've got fully initialized collection after method execution. Now, `FirstPageFunc` will be called only during enumeration. Is that expected?

Reply:
Well, this is not expected. Do you have a suggestion how we could restore previous behavior? Does ``PageableHelpers` have related APIs to do this? Like I first call the API and get the array then I pass it into some function and it gives me a `Pageable` of something?

Reply:
Sure. You can use `FromPages`:
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pageable.cs#L97
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/AsyncPageable.cs#L118
e.g.:
```cs
using var scope = _clientDiagnostics.CreateScope("VirtualMachineScaleSetVirtualMachineExtensionCollection.GetAll");
scope.Start();
try
{
    var response = _virtualMachineScaleSetVMExtensionsRestClient.List(Id.SubscriptionId, Id.ResourceGroupName, Id.Parent.Name, Id.Name, expand, cancellationToken: cancellationToken);
    var page = Page.FromValues(response.Value.Value.Select(value => new VirtualMachineScaleSetVirtualMachineExtension(Parent, value)), null, response.GetRawResponse());
    return Pageable<T>.FromPages(new[]{page});
}
catch (Exception e)
{
    scope.Failed(e);
    throw;
}
```

Reply:
Thank you! Let me switch to this and confirm with Michael

Reply:
I suppose this should be expected. We could merge with this behaviour and if this is not good enough, we could revisit this piece of logic. Changing this is not a breaking change for the SDK API.